test/puppetlabs/certificate_authority/core_test.clj

```diff
@@ -169,6 +169,21 @@
           cn (get-cn-from-x500-principal x500-principal)]
       (is (= "myagent" cn)))))
 
+(deftest cn-from-x509-certificate-test
+  (testing "cn extracted from an X509Certificate"
+    (let [subject         (cn "foo")
+          key-pair        (generate-key-pair)
```

Comment:
Generating the keypairs is what makes the unit tests so slow! We've been passing `512` as an argument when it doesn't matter just to cut down on the time it takes.  Would you mind specifying that with these two calls?